src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,641 @@
+            // the debt at time of cull
+            daiDebt = vat.gem(ilk, address(pool));
+        } else {
+            // MCD caged
+            // debt is obtained from free collateral owned by the End module
+            end_ = end;
+            end_.skim(ilk, address(pool));
+            daiDebt = vat.gem(ilk, address(end_));
+        }
+
+        uint256 availableAssets = pool.maxWithdraw();
+
+        // Unwind amount is limited by how much:
+        // - max reduction desired
+        // - assets available
+        // - dai debt tracked in vat (CDP or free)
+        uint256 amount = _min(
+                            _min(
+                                _min(
+                                    supplyReduction,
+                                    availableAssets
+                                ),
+                                daiDebt
+                            ),
+                            uint256(type(int256).max)
+                         );
+
+        // Determine the amount of fees to bring back
+        uint256 fees = 0;
+        if (assetBalance > daiDebt) {
+            unchecked {
+                fees = assetBalance - daiDebt;
+            }
+
+            if (amount + fees > availableAssets) {
+                unchecked {
+                    fees = availableAssets - amount;
+                }
+            }
+        }
+
+        if (amount == 0 && fees == 0) {
+            emit Unwind(ilk, 0, 0);
+            return;
+        }
+
+        // To save gas you can bring the fees back with the unwind
+        uint256 total = amount + fees;
+        require(pool.withdraw(total), "DssDirectDepositHub/withdraw-failed");
+        daiJoin.join(address(this), total);
+
+        // normalized debt == erc20 DAI to pool (Vat rate for this ilk fixed to 1 RAY)
+
+        if (mode == Mode.NORMAL) {
+            vat.frob(ilk, address(pool), address(pool), address(this), -int256(amount), -int256(amount));
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(this), vow, fees * RAY);
+        } else if (mode == Mode.MODULE_CULLED) {
+            vat.slip(ilk, address(pool), -int256(amount));
+            vat.move(address(this), vow, total * RAY);
+        } else {
+            // This can be done with the assumption that the price of 1 collateral unit equals 1 DAI.
+            // That way we know that the prev End.skim call kept its gap[ilk] emptied as the CDP was always collateralized.
+            // Otherwise we couldn't just simply take away the collateral from the End module as the next line will be doing.
+            vat.slip(ilk, address(end_), -int256(amount));
+            vat.move(address(this), vow, total * RAY);
+        }
+
+        emit Unwind(ilk, amount, fees);
+    }
+
+    /**
+        @notice remove nope on daiJoin to kill hub
+    */
+    function nope() external auth {
+        vat.nope(address(daiJoin));
+    }
```

Comment:
Please remove, it is useless. We need to avoid adding things that are not really needed. For example I had to review if malicious governance could actually use it to block the `Hub` to work correctly as there is not `vat.live` protection. Finally it isn't the case as it will just block `daiJoin.exit` which only executes in `_wind` which doesn't happen after shutdown. So we should be fine, but it is an example of why we need to avoid unnecessary functions.

Reply:
done f5d7fca